libs/language-server/src/lib/validation/checks/constrainttype-specific/property-assignment.ts

```diff
@@ -0,0 +1,79 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import {
+  EvaluationContext,
+  PrimitiveValuetypes,
+  PropertyAssignment,
+  evaluatePropertyValue,
+} from '../../../ast';
+import { PropertySpecification } from '../../../meta-information';
+import { ValidationContext } from '../../validation-context';
+
+export function checkConstraintTypeSpecificProperties(
```

Comment:
What do you mean by "repeated"?
`property-assigment.ts` deals with validations of single property assignments.
`property-body.ts` deals with validations of a completely property body which means between property assignments.

We might join both files and have two different top-level `check` functions. 
But that breaks the pattern we use for validations: one file per grammatical entity.
So I'd rather keep it like that.

Btw, the `property-assignment` files will vanish as soon as we move the constraint executors to the language-server. Then we don't need those anymore (if we can express them with value types).

Reply:
Ah yeah that makes sense. I totally missed that checking min > max is only possible if you know both values of course. My bad and this makes sense as it is.

Reply:
I'm also not too happy with that implicit knowledge here to understand it. But I think it will be solved by self-describing the constraints via valuetypes.